osbrain/tests/test_agent.py

```diff
@@ -105,6 +105,88 @@ def test_agent_shutdown(nsproxy):
     assert 'a0' not in nsproxy.list()
 
 
+def test_close_socket_agent_shutdown(nsproxy):
+    """
+    Check that the socket is closed and the socket file removed after the agent
+    is shut down
+    """
+    agent = run_agent('name')
+    address = agent.bind('PUSH')
+    agent.shutdown()
+    assert not address.address.exists()
+
+
+def test_close_socket_agent_kill(nsproxy):
+    """
+    Check that the socket is closed and the socket file removed after the agent
+    is killed
+    """
+    agent = run_agent('name')
+    address = agent.bind('PUSH')
+    agent.kill()
+    assert not address.address.exists()
+
+
+def test_close_socket_agent_blocked(nsproxy):
+    """
+    Check that the socket is closed and the socket file removed when a blocked
+    agent is requested to shut down
+    """
+    def log(agent, msg):
+        time.sleep(3)
+        return msg
+
+    alice = run_agent('alice')
+    bob = run_agent('bob')
+
+    addr = alice.bind('REP', alias='main', handler=log)
+    bob.connect(addr, alias='main')
+
+    Timer(1, nsproxy.shutdown).start()
+    bob.send('main', 'Hello world')
+    rep = bob.recv('main')
+    print(rep)
+
+    assert not addr.address.exists()
+
+
+def test_close_socket_agent_crash(nsproxy):
+    """
+    Check that the socket is closed and the socket file removed after the agent
+    crashes
+    """
+    agent = run_agent('agent')
+    addr = agent.bind('PUSH', 'main')
+    with pytest.raises(RuntimeError):
+        agent.raise_exception()
+
+    assert not addr.address.exists()
+
+
+def test_close_socket_nameserver_shutdown(nsproxy):
```

Comment:
Maybe all tests should start with `test_close_agent_socket_xxx` (even if `xxx` is `nameserver_shutdown`).

Reply:
Makes sense.